thangs_importer.py

```diff
@@ -0,0 +1,303 @@
+            return
+
+        r = requests.get(modelURL, stream=True)
+        self.uid = str(model_title)
+        temp_dir = os.path.join(Config.THANGS_MODEL_DIR, self.uid)
+        if not os.path.exists(temp_dir):
+            os.makedirs(temp_dir)
+
+        print("Temp Dir")
+        print(temp_dir)
+        # archive_path = os.path.join(temp_dir, '{}.zip'.format(uid))
+        archive_path = os.path.join(temp_dir, '{}.stl'.format(self.uid))
+        print("Arch Path")
+        print(archive_path)
+        if not os.path.exists(archive_path):
+            print("Starting Count")
+            wm = bpy.context.window_manager
+            wm.progress_begin(0, 100)
+            with open(archive_path, "wb") as f:
+                total_length = r.headers.get('content-length')
+                if total_length is None:  # no content length header
+                    f.write(r.content)
+                else:
+                    dl = 0
+                    total_length = int(total_length)
+                    for data in r.iter_content(chunk_size=4096):
+                        dl += len(data)
+                        f.write(data)
+                        done = int(100 * dl / total_length)
+                        wm.progress_update(done)
+
+            wm.progress_end()
+        else:
+            print('Model already downloaded')
+
+        # gltf_path, gltf_zip = unzip_archive(archive_path)
+        self.gltf_path = archive_path
+        self.import_model()
+        return
+        if self.gltf_path:
+            try:
+                print("Starting Import Thread")
+                # self.import_thread = threading.Thread(
+                #    target=self.import_model).start()
+                self.import_model()
+                # import_model(gltf_path, uid)
+            except Exception as e:
+                import traceback
+                print(traceback.format_exc())
+        else:
+            print("ERROR", "Download error",
+                  "Failed to download model (url might be invalid)")
+        return
+
+    def unzip_archive(self, archive_path):
+        if os.path.exists(archive_path):
+            import zipfile
+            try:
+                zip_ref = zipfile.ZipFile(archive_path, 'r')
+                extract_dir = os.path.dirname(archive_path)
+                zip_ref.extractall(extract_dir)
+                zip_ref.close()
+            except zipfile.BadZipFile:
+                print('Error when dezipping file')
+                os.remove(archive_path)
+                print('Invaild zip. Try again')
+                return None, None
+
+            stl_file = os.path.join(extract_dir, 'scene.gltf')
+            return stl_file, archive_path
+
+        else:
+            print('ERROR: archive doesn\'t exist')
+
+    def import_model(self):
+        #old_objects = [o.name for o in bpy.data.objects]
+        print("Starting File Import")
+        bpy.ops.import_mesh.stl(filepath=self.gltf_path)
```

Comment:
We're always importing as stl, but the path variable is `gltf_path`.  Should we make that something like `model_path` instead?